core/src/main/java/com/linecorp/armeria/common/AnnotatedRequest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2025 LY Corporation
+ *
+ * LY Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
```

Comment:
Shouldn't this be in the server package?

Reply:
I was imagining client-side request/responses will also be recorded here. Do you prefer this class is moved to the server package?

Reply:
> What do you also think of including the method name as we do in the gRPC service?

I could, but what benefit would it bring?

If logging the method name is the objective, would it make sense to leave the method name at `RequestLog#name` instead so that both metrics/logging consistently view the same value?

Reply:
> I could, but what benefit would it bring?

Unifying the logic for setting `RequestLog#name`?

Thrift and gRPC services currently use `RpcRequest` to set the `RequestLog#name`:
https://github.com/line/armeria/blob/79112f537eb851aa2aba3b534777a608f2c71e0d/core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java#L1149-L1155

I was thinking we could generalize this by:
- Introducing a new `ApplicationRequest` class
- Using `ApplicationRequest` instead of `RpcRequest` for setting method names, parameters, etc.

Reply:
@minwoox suggested that `AnnotatedRequest`/`AnnotatedResponse` is moved to the internal package so that we can possibly generalize this in the future

Reply:
Updated to just remove the interfaces `AnnotatedRequest`/`AnnotatedResponse` and made them package-private.